modules/UpgradeWizard/index.php

```diff
@@ -497,6 +499,73 @@ function getSelectedModulesForLayoutMerge()
 </script>
 eoq;
 
+// If we are on step  4 then force a redirect to run again to pick up changes to smarty before setting the template.
+if ($_REQUEST['step'] === 4 && !$_REQUEST['additional_step']) {
+    // Set session variables
+    $_SESSION['UW_MAIN'] = $uwMain;
+    $_SESSION['UW_JS'] = $js;
+    $_SESSION['STEP'] = $step;
+    $_SESSION['STEPS'] = $steps;
+    $_SESSION['MOD'] = $mod_strings;
+    $_SESSION['APP'] = $app_strings;
+    $_SESSION['showNext'] = $showNext;
+    $_SESSION['showCancel'] = $showCancel;
+    $_SESSION['showBack'] = $showBack;
+    $_SESSION['showRecheck'] = $showRecheck;
+    $_SESSION['showDone'] = $showDone;
+    $_SESSION['showExit'] = $showExit;
+    $_SESSION['STEP_NEXT'] = $stepNext;
+    $_SESSION['STEP_CANCEL'] = $stepCancel;
+    $_SESSION['STEP_BACK'] = $stepBack;
+    $_SESSION['STEP_RECHECK'] = $stepRecheck;
+    $_SESSION['STEPS_FILE'] = $steps['files'];
+    $_SESSION['UW_HISTORY'] = $uwHistory;
+    $_SESSION['disableNextForLicense'] = $disableNextForLicense;
+    $u_allow = 'true';
+    if (isset($stop) && $stop === true) {
+        $frozen = (isset($frozen)) ? '<br />' . $frozen : '';
+        $_SESSION['frozen'] = $frozen;
+        if ($step === 'upload') {
+            $u_allow = 'false';
+        }
+    }
+    $_SESSION['u_allow'] = $u_allow;
+    if (!empty($GLOBALS['top_message'])) {
+        $_SESSION['top_message'] = $GLOBALS['top_message'];
+    }
+
+    // Redirect to additional step and kill the process
+    header('Location: ' . $_SERVER['HTTP_REFERER'] . '?module=UpgradeWizard&action=index&step=4&additional_step=true');
+    die();
```

Comment:
Ideally we'd want to avoid introducing new ```die()``` in the code to avoid increasing the complexity of unit testing but in this particular case I don't see too many issues doing it this way.